unicodetools/org/unicode/tools/emoji/EmojiData.java

```diff
@@ -1344,35 +1381,29 @@ public String normalizeVariant(String emojiSequence) {
 	boolean didMultiples = false;
 	if (addMultiples) {
 	    int handshakePos = singletonOrSequence.indexOf(ZWJ_HANDSHAKE_ZWJ);
-	    if (handshakePos > 0 && version.compareTo(Emoji.VERSION11) > 0) {
-		boolean afterVersion12 = version.compareTo(Emoji.VERSION12) > 0;
-		didMultiples = true;
-		// TODO HACK for now. If we add other groupings with skintones, generalize
-		String prefix = singletonOrSequence.substring(0, handshakePos);
-		String postfix = singletonOrSequence.substring(handshakePos + ZWJ_HANDSHAKE_ZWJ.length());
-		boolean sameAffix = prefix.equals(postfix);
-		for (String mod : EmojiData.MODIFIERS) {
-		    String prefixMod = addModifierPart(prefix, mod);
-		    if (prefixMod == null) {
-			throw new IllegalArgumentException("internal error");
+	    if (handshakePos > 0) {
+		if (version.compareTo(Emoji.VERSION11) > 0) {
+		    boolean afterVersion12 = version.compareTo(Emoji.VERSION12) > 0;
+		    didMultiples = true;
+		    // TODO HACK for now. If we add other groupings with skintones, generalize
+		    String prefix = singletonOrSequence.substring(0, handshakePos);
+		    String postfix = singletonOrSequence.substring(handshakePos + ZWJ_HANDSHAKE_ZWJ.length());
+		    boolean sameAffix = prefix.equals(postfix);
+		    final boolean skipIfFirstLighterThanSecond = !afterVersion12 && sameAffix;
+		    addMultiples(prefix, ZWJ_HANDSHAKE_ZWJ, postfix, skipIfFirstLighterThanSecond, output);
+		}
+	    } else if (false && MULTIPLE_SKINS.contains(singletonOrSequence)) {
+		if (version.compareTo(Emoji.VERSION13) > 0) {
```

Comment:
This if false should just be removed, but leaving it in for the moment in case I need to retract it.